src/api/posts/test/posts.test.js

```diff
@@ -0,0 +1,166 @@
+const request = require('supertest');
+const { app } = require('../src');
+const { addPost, getPost } = require('../src/storage');
+const hash = require('../../auth/src/hash');
+
+describe('/posts', () => {
+  const requestedItems = 50;
+  const maxItems = 100;
+  const createdItems = 150;
+  const nonInteger = 'test';
+
+  const posts = [...Array(createdItems).keys()].map((item) => {
+    const guid = `http://telescope${item}.cdot.systems`;
+    const id = hash(guid);
+    return {
+      id,
+      guid,
+      author: 'author',
+      title: 'title',
+      html: 'html',
+      updated: new Date('2009-09-07T22:23:00.544Z'),
+      published: new Date('2009-09-07T22:20:00.000Z'),
+      url: 'foo',
+      site: 'foo',
+    };
+  });
+
+  beforeAll(() => posts.map((post) => addPost(post)));
+
+  test('default number of items should be returned', async (done) => {
+    const res = await request(app).get('/');
+
+    expect(res.status).toBe(200);
+    expect(res.get('Content-type')).toContain('application/json');
+    expect(res.get('X-Total-Count')).toBe(createdItems.toString());
+    expect(res.body.length).toBeGreaterThan(0);
+    expect(res.body instanceof Array).toBe(true);
+    done();
```

Comment:
The way you're writing these tests, you probably don't need `done()` here, it's more when you have a callback function (e.g., if you used supertest's `end(function(...){...})` or something).  It's OK to leave it, or you can try removing too.